shared/activate.ts

```diff
@@ -79,3 +204,104 @@ function activateWithoutLSP(
+}
+
+/**
+ * Return the Sourcegraph URL from the current configuration.
+ *
+ * @param settings The user configured sourcegraph URL.
+ * @param languageID The language identifier.
+ */
+function sourcegraphURL(setting: string | undefined, languageID: string): URL {
+    const url = setting || sourcegraph.internal.sourcegraphURL.toString()
+
+    try {
+        return new URL(url)
+    } catch (err) {
+        if (err.message?.includes('Invalid URL')) {
+            console.error(
+                new Error(
+                    [
+                        `Invalid ${languageID}.sourcegraphUrl ${url} in your Sourcegraph settings.`,
+                        'Make sure it is set to the address of Sourcegraph from the perspective of the language server (e.g. http://sourcegraph-frontend:30080).',
+                        `Read the full documentation for more information: https://github.com/sourcegraph/sourcegraph-${languageID}`,
+                    ].join('\n')
+                )
+            )
+        }
+
+        throw err
+    }
+}
+
+/**
+ * When the current settings change, determine if we need to supply/revoke the
+ * externalReferencesProvider to the LSP client. This will cause the LSP client
+ * features to re-register a the references via the extension context.
+ *
+ * @param ctx The extension context.
+ * @param implementationId The identifier of the registered locations provider.
+ * @param settingsSubject An observable of settings values.
+ * @param settingName The setting name to watch for changes.
+ * @param featureOptionsSubject The feature options to funnel changes into.
+ * @param externalReferencesProvider The external references provider to register.
+ */
+function registerExternalReferenceProviderToggle<
+    S extends { [key: string]: any }
+>(
+    ctx: sourcegraph.ExtensionContext,
+    implementationId: string,
+    settingsSubject: Observable<S>,
+    settingName: string,
+    featureOptionsSubject: Subject<FeatureOptions>,
+    externalReferencesProvider: ReferencesProvider
+): void {
+    ctx.subscriptions.add(
+        from(settingsSubject)
+            .pipe(
+                distinctUntilChanged(),
+                map(v => ({
+                    implementationId,
+                    ...(v[settingName] ? { externalReferencesProvider } : {}),
+                }))
+            )
+            .subscribe(featureOptionsSubject)
+    )
+}
+
+/**
+ * Register a panel view that will hold the results from the LSP implementations provider.
+ *
+ * @param ctx The extension context.
+ * @param implementationId The identifier of the registered locations provider.
+ */
+function registerImplementationsPanel(
+    ctx: sourcegraph.ExtensionContext,
+    implementationId: string
+): void {
+    const panelView = sourcegraph.app.createPanelView(implementationId)
+    panelView.title = 'Implementations'
+    panelView.component = { locationProvider: implementationId }
+    panelView.priority = 160
```

Comment:
Why 160?

Reply:
That's what was there, man! I'm scared to change it.